container/containerd/client.go

```diff
@@ -125,6 +130,21 @@ func (c *client) Version(ctx context.Context) (string, error) {
 	return response.Version, nil
 }
 
+func (c *client) ContainerFsUsage(ctx context.Context, snapshotter, snapshotkey string) (*common.FsUsage, error) {
+	usage, err := c.snapshotsService.Usage(ctx, &snaptshotapi.UsageRequest{
```

Comment:
how expensive is this?

will containerd snapshotter have to recaculate the usage every time this is called or does it cache the usage internally?

Reply:
Containerd does not cache it  and will recaculate again when we send a request,but we update it at a fixed frequency, see `realFsHandler.trackUsage` in `cadvisor/container/common/fsHandler.go`, so I think it's ok .

And compare with the scenario of docker, it's same,the containerd only caculate the usage of upper(or diff)  directory,not the rootfs (merged).

Reply:
Yes, there is a cache in it's CRI implemention . But `Usage` method will recaculate the usage every time, I have change it to `ContainerStats`